tests/test_extractor.py

```diff
@@ -30,3 +32,110 @@ def test_multiple_chunks(self, tmp_path: Path):
         assert sorted(tmp_path.iterdir()) == [written_path1, written_path2]
         assert written_path1.read_bytes() == content[:4]
         assert written_path2.read_bytes() == content[4:]
+
+
+def test_fix_permission(tmpdir: Path):
```

Comment:
I would add a testcase to see if `fix_extracted_directory` is fixing permissions recursively

an interesting edge case:

testdir/ 0o200
testdir/testdir2/ 0x000
testdir/testdir2/testfile 0o200

Reply:
The `test_fix_extracted_directory` test takes care of this. That's a nice edge case, had to fix the function :)